tests/utils.py

```diff
@@ -320,7 +320,19 @@ def get_proxy(self):
         proxy.start()
         return proxy
 
-    def generate_block(self, numblocks=1):
+    # wait_for_mempool can be used to wait for the mempool before generating blocks:
+    # True := wait for at leasst 1 transations
+    # int > 0 := wait for at least N transactions
+    # 'tx_id' := wait for one transaction id given as a string
+    # ['tx_id1', 'tx_id2'] := wait until all of the specified transaction IDs
+    def generate_block(self, numblocks=1, wait_for_mempool=0):
+        if wait_for_mempool:
+            if isinstance(wait_for_mempool, str):
```

Comment:
Is there any difference between `wait_for_mempool=True` and `wait_for_mempool=1`?

Reply:
> Is there any difference between `wait_for_mempool=True` and `wait_for_mempool=1`?

According to what criteria ? If this is about how they will evaluate line 329, they will both pass the condition. Otherwise there is a big difference : it changes the type of the `wait_for_mempool` variable (/object) :
```python
<class 'bool'>
>>> type(1)
<class 'int'>
```

Reply:
nope, both will wait for at least one TX